### PR TITLE
Show and log a warning when schema contains '.'.

### PR DESCRIPTION
**Pull request by ueshin**
### Description

Shows and logs a warning when schema contains `'.'`.

As a side work, trying to help more cases where schema contains `'.'` in `list_relations_without_caching`.

### CODE REVIEW

File: dbt/adapters/databricks/connections.py [hunk 1]
```diff
@@ -84,6 +84,13 @@ def __pre_deserialize__(cls, data: Dict[Any, Any]) -> Dict[Any, Any]:
         return data
 
     def __post_init__(self) -> None:
+        if "." in self.schema:
+            logger.warning(
+                "The specified schema contains '.', which could cause unexpected behavior.\n"
```
Comment (allisonwang-db) [inline dbt/adapters/databricks/connections.py:89]:
```suggestion
               f"The specified schema '{self.schema}' contains '.', which could cause unexpected behavior.\n"
```

File: dbt/adapters/databricks/connections.py [hunk 2]
```diff
@@ -84,6 +84,13 @@ def __pre_deserialize__(cls, data: Dict[Any, Any]) -> Dict[Any, Any]:
...
+                "It will not be allowed in the future release.\n"
+                "If you are trying to set a catalog, use `catalog` instead.\n"
```
Comment (allisonwang-db) [inline dbt/adapters/databricks/connections.py:91]:
```suggestion
                "If you are trying to set a catalog, please use `catalog` instead.\n"
```

**Review by allisonwang-db — APPROVED** [review]
Thanks for adding this!

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**